_docs/setup/install-kubernetes.md

```diff
@@ -231,52 +200,34 @@ kubectl create -f <(istioctl kube-inject -f <your-app-spec>.yaml)
 
 ## Uninstalling
 
-You can use the [Istio Helm chart](https://github.com/kubernetes/charts/tree/master/incubator/istio#uninstalling-the-chart) to uninstall, or follow the steps below.
-
-1. Uninstall Istio core components:
+1. If you installed Istio addons, uninstall them:
 
-   * If Istio was installed without Istio auth feature:
-
-     ```bash
-     kubectl delete -f install/kubernetes/istio.yaml
-     ```
-
-   * If Istio was installed with auth feature enabled:
+   ```bash
+   kubectl delete -f install/kubernetes/addons/
+   ```
 
-     ```bash
-     kubectl delete -f install/kubernetes/istio-auth.yaml
-     ```
 
-2. Uninstall RBAC Istio roles:
+1. Uninstall Istio core components. For the {{ site.data.istio.version }} release, the uninstall will delete the RBAC permissions, the `istio-system` namespace, and hierarchically all resources under it.
+   It is safe to ignore errors for non-existent resources, since they may have been deleted hierarchically.
 
-   * If beta version was installed:
+   * If Istio was installed with mutual TLS authentication disabled:
 
      ```bash
-     kubectl delete -f install/kubernetes/istio-rbac-beta.yaml
+     kubectl delete -f install/kubernetes/istio.yaml
      ```
 
-   * If alpha version was installed:
+   * If Istio was installed with mutual TLS authentication enabled:
 
      ```bash
-     kubectl delete -f install/kubernetes/istio-rbac-alpha.yaml
+     kubectl delete -f install/kubernetes/istio-auth.yaml
      ```
 
-3. If you installed Istio addons, uninstall them:
-
-   ```bash
-   kubectl delete -f install/kubernetes/addons/
-   ```
-   and, if Jaeger was installed,
-   ```bash
-   kubectl delete -f https://raw.githubusercontent.com/jaegertracing/jaeger-kubernetes/master/all-in-one/jaeger-all-in-one-template.yml
-   ```
-
-
-4. Delete Istio Kubernetes [TPRs](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-third-party-resource):
+1. Delete Istio's [Kubernetes CRDs](https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/):
 
    ```bash
-   kubectl delete istioconfigs --all
-   kubectl delete thirdpartyresource istio-config.istio.io
+   kubectl delete crd destinationpolicies.config.istio.io
+   kubectl delete crd egressrules.config.istio.io
+   kubectl delete crd routerules.config.istio.io
```

Comment:
are you sure the above 3 is exhaustive/enough ? seems like just pilot ? this is what I have in my cluster
```
NAME                                  KIND
attributemanifests.config.istio.io    CustomResourceDefinition.v1beta1.apiextensions.k8s.io
checknothings.config.istio.io         CustomResourceDefinition.v1beta1.apiextensions.k8s.io
deniers.config.istio.io               CustomResourceDefinition.v1beta1.apiextensions.k8s.io
destinationpolicies.config.istio.io   CustomResourceDefinition.v1beta1.apiextensions.k8s.io
egressrules.config.istio.io           CustomResourceDefinition.v1beta1.apiextensions.k8s.io
listcheckers.config.istio.io          CustomResourceDefinition.v1beta1.apiextensions.k8s.io
listentries.config.istio.io           CustomResourceDefinition.v1beta1.apiextensions.k8s.io
logentries.config.istio.io            CustomResourceDefinition.v1beta1.apiextensions.k8s.io
memquotas.config.istio.io             CustomResourceDefinition.v1beta1.apiextensions.k8s.io
metrics.config.istio.io               CustomResourceDefinition.v1beta1.apiextensions.k8s.io
noops.config.istio.io                 CustomResourceDefinition.v1beta1.apiextensions.k8s.io
prometheuses.config.istio.io          CustomResourceDefinition.v1beta1.apiextensions.k8s.io
quotas.config.istio.io                CustomResourceDefinition.v1beta1.apiextensions.k8s.io
reportnothings.config.istio.io        CustomResourceDefinition.v1beta1.apiextensions.k8s.io
routerules.config.istio.io            CustomResourceDefinition.v1beta1.apiextensions.k8s.io
rules.config.istio.io                 CustomResourceDefinition.v1beta1.apiextensions.k8s.io
stackdrivers.config.istio.io          CustomResourceDefinition.v1beta1.apiextensions.k8s.io
statsds.config.istio.io               CustomResourceDefinition.v1beta1.apiextensions.k8s.io
stdios.config.istio.io                CustomResourceDefinition.v1beta1.apiextensions.k8s.io
svcctrls.config.istio.io              CustomResourceDefinition.v1beta1.apiextensions.k8s.io
```

Reply:
this works for instance for me:
```
$ kubectl get crd -o 'jsonpath={.items[*].metadata.name}'|grep config.istio.io |xargs kubectl delete crd 
customresourcedefinition "attributemanifests.config.istio.io" deleted
customresourcedefinition "checknothings.config.istio.io" deleted
customresourcedefinition "deniers.config.istio.io" deleted
customresourcedefinition "destinationpolicies.config.istio.io" deleted
customresourcedefinition "egressrules.config.istio.io" deleted
customresourcedefinition "listcheckers.config.istio.io" deleted
customresourcedefinition "listentries.config.istio.io" deleted
customresourcedefinition "logentries.config.istio.io" deleted
customresourcedefinition "memquotas.config.istio.io" deleted
customresourcedefinition "metrics.config.istio.io" deleted
customresourcedefinition "noops.config.istio.io" deleted
customresourcedefinition "prometheuses.config.istio.io" deleted
customresourcedefinition "quotas.config.istio.io" deleted
customresourcedefinition "reportnothings.config.istio.io" deleted
customresourcedefinition "routerules.config.istio.io" deleted
customresourcedefinition "rules.config.istio.io" deleted
customresourcedefinition "stackdrivers.config.istio.io" deleted
customresourcedefinition "statsds.config.istio.io" deleted
customresourcedefinition "stdios.config.istio.io" deleted
customresourcedefinition "svcctrls.config.istio.io" deleted
```